### PR TITLE
Implement diffForHumans

### PR DESCRIPTION
**Pull request by markstory**
This shores up a big incompatibility between Carbon and Chronos. I've only implemented english, as I feel that is the minimal set of messages we need for compatibility. I also just didn't want to add any dependencies to chronos.

Refs cakephp/cakephp#7713

### CODE REVIEW

File: src/Traits/DifferenceTrait.php [hunk 1]
```diff
@@ -234,4 +237,98 @@ public static function fromNow($datetime)
...
+        if (static::translator()->exists($tryKeyExists)) {
+            $time = static::translator()->plural($tryKeyExists, $count, array('count' => $count));
+        }
+        return static::translator()->singular($transId, array('time' => $time));
+    }
+
+    /**
+     * Get the translator instance or overwrite the current one.
+     *
+     * @return Cake\Chronos\Translator The translator instance.
```
Comment (dereuromark) [inline src/Traits/DifferenceTrait.php]: |null

Comment (markstory) [inline src/Traits/DifferenceTrait.php]: Thanks.

**Comment by HavokInspiration** [comment]
This is looking good.
I fixed a bunch of CS warnings.

How do you suggest we provide other languages than english ?

**Comment by lorenzo** [comment]
I was more inclined to see the implementation of the diff inside another object. Do you think that would make sense?

File: src/ChronosInterface.php [hunk 2]
```diff
@@ -843,6 +843,31 @@ public function subSecond($value = 1);
...
+     *
+     * When comparing a value in the future to another value:
+     * 1 hour after
+     * 5 months after
+     *
+     * @param \Cake\Chronos\ChronosInterface|null $other The datetime to compare with.
+     * @param bool $absolute removes time difference modifiers ago, after, etc
+     * @return string
+     */
+    public function diffForhumans(ChronosInterface $other = null, $absolute = false);
```
Comment (HavokInspiration) [inline src/ChronosInterface.php]: Shouldn't it be ``diffForHumans`` ?

Comment (dereuromark) [inline src/ChronosInterface.php]: It should.
I would also uppercase the removes: `bool $absolute Removes`

**Comment by markstory** [comment]
@lorenzo Ok. That might make it easier to integrate CakePHP's i18n subsystem.

**Comment by markstory** [comment]
@lorenzo I've split out the formatter, let me know what you think.

**Comment by lorenzo** [comment]
Good job @markstory!

**Pull request merged**